zenoh-kotlin/src/commonMain/kotlin/io/zenoh/Config.kt

```diff
@@ -16,14 +16,20 @@ package io.zenoh
 
 import java.io.File
 import java.nio.file.Path
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.encodeToJsonElement
+
 
 /**
  * Config class to set the Zenoh configuration to be used through a [Session].
  *
  * @property path The path to the configuration file.
  * @constructor Create empty Config
  */
-class Config private constructor(internal val path: Path? = null) {
+class Config private constructor(internal val path: Path? = null, internal val jsonConfig: JsonElement? = null) {
 
```

Comment:
I would add a constructor from `string` likewise we do in Rust and other bindings => https://github.com/eclipse-zenoh/zenoh-python/blob/master/zenoh/config.py#L37
Then how this string is read/constructed is up to the users.



(I.e., it could be retrieved via an API and just passed to create a Zenoh session).

Reply:
I added from json. There is no official support for json5 in kotlin, but there are third party libraries. Do you think we should make the project depend on those libraries? I opted for not, but I wonder what you think.

Reply:
To me it is fine like it is now, no need to add a dependency.